From: Ben Hutchings <ben@decadent.org.uk>
Date: Mon, 19 Mar 2018 20:53:19 +0000
Subject: x86/cpu: Convert vulnerabilities attributes
Forwarded: not-needed

The backport of commit 61dc0f555b5c ("x86/cpu: Implement CPU
vulnerabilites sysfs functions") to 3.2-stable added attributes for
CPU vulnerabilities, as sysdev_class_attributes.  wheezy has
additional backported changes to convert CPU sysdevs into regular
devices, so we need to convert these attributes back to
device_attributes like they are upstream.

---
--- a/arch/x86/kernel/cpu/bugs.c
+++ b/arch/x86/kernel/cpu/bugs.c
@@ -443,8 +443,7 @@ retpoline_auto:
 #undef pr_fmt
 
 #ifdef CONFIG_SYSFS
-ssize_t cpu_show_meltdown(struct sysdev_class *dev,
-			  struct sysdev_class_attribute *attr, char *buf)
+ssize_t cpu_show_meltdown(struct device *dev, struct device_attribute *attr, char *buf)
 {
 	if (!boot_cpu_has_bug(X86_BUG_CPU_MELTDOWN))
 		return sprintf(buf, "Not affected\n");
@@ -453,16 +452,14 @@ ssize_t cpu_show_meltdown(struct sysdev_
 	return sprintf(buf, "Vulnerable\n");
 }
 
-ssize_t cpu_show_spectre_v1(struct sysdev_class *dev,
-			    struct sysdev_class_attribute *attr, char *buf)
+ssize_t cpu_show_spectre_v1(struct device *dev, struct device_attribute *attr, char *buf)
 {
 	if (!boot_cpu_has_bug(X86_BUG_SPECTRE_V1))
 		return sprintf(buf, "Not affected\n");
 	return sprintf(buf, "Mitigation: __user pointer sanitization\n");
 }
 
-ssize_t cpu_show_spectre_v2(struct sysdev_class *dev,
-			    struct sysdev_class_attribute *attr, char *buf)
+ssize_t cpu_show_spectre_v2(struct device *dev, struct device_attribute *attr, char *buf)
 {
 	if (!boot_cpu_has_bug(X86_BUG_SPECTRE_V2))
 		return sprintf(buf, "Not affected\n");
